app/models/requests/DataRequest.scala

```diff
@@ -19,15 +19,26 @@ package models.requests
 import models.{LoggedInUser, UserAnswers}
 import play.api.mvc.{Request, WrappedRequest}
 
+abstract class RequiredSchemeDataRequest[A](request: Request[A]) extends WrappedRequest[A](request) {
```

Comment:
Could you explain the purpose of this? Seems to be used in the views only. Might be worth going thru this with team